packages/runtime/src/state/actions.ts

```diff
@@ -616,12 +621,15 @@ export function messageBuilderPropController<T>(
   }
 }
 
-export function changeApiResource(resource: APIResource): ChangeAPIResourceAction {
-  return { type: ActionTypes.CHANGE_API_RESOURCE, payload: { resource } }
+export function changeApiResource(
+  resource: APIResource,
+  locale?: string | null,
```

Comment:
Here and below, keeping the `locale` argument optional for now for backward compatibility.